nifi-commons/nifi-utils/src/test/java/org/apache/nifi/util/validator/TestMockValidator.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.util.validator;
+
+import org.apache.nifi.components.ValidationContext;
+import org.apache.nifi.components.ValidationResult;
+import org.apache.nifi.components.Validator;
+
+/**
+ * TestMockValidator wraps a {@class Validator} and provides statistics on it's interactions.
+ * Because many of the {@class Validator} instances returned from {@class StandardValidator }
+ * are not mockable with with mockito, this is required to know, when running a test, if a
+ * {@class Validator} was in fact called, for example.
+ */
+public class TestMockValidator implements Validator {
```

Comment:
Do you think this would be useful in general for unit tests? If so perhaps it should be in the `nifi-mock` module rather than the test sources for `nifi-utils`? I'm fine either way, but this looks like it could be useful in other places, especially things like unit tests for scripted components.

Reply:
That was my first thought @mattyb149 , but I thought it would introduce dependency issues.  Maybe I was wrong?

Reply:
I *think* the ideal place would be mocks.  The mocks brings in utils and api and utils uses mocks to test part gave me pause

Reply:
I could try it in mocks and "see" what happens, then rebase/force push if it works, how does that sound?

Reply:
@mattyb149, I tried to move it, and add a test stop dependency to nifi-mock to nifi-utils, I get cyclic dependency errors and cannot build

Reply:
Makes sense, since the test is in nifi-utils (along with the unit-under-test). Still it sounds useful in general, what about a StandardValidators-specific one in nifi-utils (the way you have it now), perhaps having `StandardValidator` in the name (I was thinking about something besides `Test` that indicates you're instrumenting a real one, perhaps `InstrumentedStandardValidator` for the one in nifi-utils, and then have another one in nifi-mock called `InstrumentedValidator`?

Reply:
As long as everyone is OK with the DRYness ;)

Reply:
Also, breaking the current naming scheme

Reply:
I'm OK with it in this instance, since one is specifically for the provided StandardValidators and the other is for custom validators. Pretty sure we've done things like this before for the same reason (search for `MockDBCPService` in the codebase 😄 )